### DISCUSSION TITLE
Deployment status badge

### DISCUSSION THREAD

**Discussion by iinfin** [vercel/community#1407] [Help]
I think if could be quite useful to have a dynamic badge for deployment status with relevant states like `queued`, `building`, `ready`, and `error` like in [Netlify](https://docs.netlify.com/monitor-sites/status-badges) (which is also available on [shieds.io](https://shields.io/category/build))

Maybe something like `https://vercel.com/badge/alias` - `https://vercel.com/badge/customdomain.com`? (I'm mentioning these samples because I thought simple alias and domain name would be more clean than UID for each project)

I see some [deployment information in API docs](https://vercel.com/docs/api#endpoints/deployments/list-deployments/response-parameters) but I'm not sure if that'd be enough for [shields.io](https://github.com/badges/shields)? In that case I guess I should open an issue on their side - for something like `https://img.shields.io/vercel/swr`.

What do you think?

**Comment by styfle** [comment]
We are tracking this feature request internally as Story 18407.

In the meantime, the deployment status can be found in GitHub's UI and GitHub's API.

See [this comment](https://github.com/vercel/vercel/discussions/4501#discussioncomment-19401) for more info.

Also see the [Vercel API docs](https://vercel.com/docs/api#endpoints/deployments/get-a-single-deployment) to read a deployment status given a deployment URL.

**Reply by iinfin** [reply]
Yes, that is very nice and informative but I was wondering about badges in particular to use in README files.
I'll look into API docs you linked, perhaps it'll be enough to be used on shields.io side. Thanks!

**Comment by hkennyv** [comment]
+1, I'd love to see something like this similar to the [Netlify deploy badges](https://www.netlify.com/blog/2019/01/29/sharing-the-love-with-netlify-deployment-badges/)

**Comment by ryota-murakami** [comment]
I really want to this badge!

**Comment by MaticSulc** [comment]
+1, would be a great feature!

**Comment by yonifra** [comment]
I would also love to see a Vercel badge!

**Comment by dytra** [comment]
i really want this so bad
currently I'm using 2 CI, vercel and travis
vercel to deploy, travis to  put the deploy status badge in my readme file
not very effective :(

**Comment by kduprey** [comment]
Is there any status update regarding this?

Currently I am using Netlify's [Status Badge](https://docs.netlify.com/monitor-sites/status-badges/#app), but would be nice to see this on this platform as well.

**Comment by therealsujitk** [comment]
I built this small application: [therealsujitk/vercel-badge](https://github.com/therealsujitk/vercel-badge/blob/main/README.md) to display if the deployment was successful or not, I hope people find it useful :innocent:.

### Example

 ![Vercel](http://therealsujitk-vercel-badge.vercel.app/?app=therealsujitk-vercel-badge)

```md
![Vercel](http://therealsujitk-vercel-badge.vercel.app/?app=therealsujitk-vercel-badge)
```

**Reply by therealsujitk** [reply]
I've added [a small interface](https://therealsujitk-vercel-badge.vercel.app) to generate your badge quickly!

**Comment by jakejarvis** [comment]
If your Vercel site is auto-deployed on GitHub pushes you can also use [shields.io](https://shields.io/) and the GitHub deployment status option with the "production" environment. Not perfect but it does work. :)

![GitHub deployments](https://img.shields.io/github/deployments/jakejarvis/jarv.is/production?label=vercel&logo=vercel&logoColor=white)

<img width="841" alt="Screen Shot 2021-06-11 at 12 28 19 PM" src="https://user-images.githubusercontent.com/1703673/121718642-83400d80-cab0-11eb-913f-676063ef9d6e.png">

**Reply by timthedev07** [reply]
It works, thanks!

**Reply by sahilofficial671** [reply]
Worked for now, thanks
